src/com/xilinx/rapidwright/edif/BinaryEDIFReader.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * 
+ * Copyright (c) 2022 Xilinx, Inc. 
+ * All rights reserved.
+ *
+ * Author: Chris Lavin, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright. 
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+/**
+ * 
+ */
+package com.xilinx.rapidwright.edif;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.xilinx.rapidwright.util.FileTools;
+
+/**
+ * A Reader for the RapidWright Binary EDIF Format
+ *  
+ * Provides a binary alternative to textual EDIF that is ~10-15X smaller and loads 5-10X faster.
+ * This is intended as a cached version of text-based EDIF as it cannot be read by Vivado.  One
+ * additional tradeoff is that it takes about 2.5-3X longer to write than text-based EDIF.
```

Comment:
Maybe this is clearer wording?
```suggestion
 * This is intended as a RapidWright-only cache (with the extension *.bedf) of this text-based
 * EDIF and cannot be read by Vivado.  One additional tradeoff is that it takes about 2.5-3X
 * longer to write than text-based EDIF, but it is expected that this binary alternative will
 * be written once and read many times.
```